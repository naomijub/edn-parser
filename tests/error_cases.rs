#![allow(missing_docs)]

use edn_parser::edn_parse;
use insta::assert_snapshot;

#[test]
fn unterminated_string() {
    let error = edn_parse("\"hello world").unwrap_err();
    let error_json = serde_json::to_string_pretty(&error).unwrap();
    assert_snapshot!(error_json);
}

#[test]
fn float_num_rational_not_supported() {
    let error = edn_parse("42.3/12").unwrap_err();
    let error_json = serde_json::to_string_pretty(&error).unwrap();
    assert_snapshot!(error_json);
}

#[test]
fn float_denom_rational_not_supported() {
    let error = edn_parse("42/12.3").unwrap_err();
    let error_json = serde_json::to_string_pretty(&error).unwrap();
    assert_snapshot!(error_json);
}
