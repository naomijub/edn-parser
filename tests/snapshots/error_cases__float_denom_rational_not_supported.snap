---
source: tests/error_cases.rs
expression: error_json
---
[
  {
    "severity": "Error",
    "code": null,
    "message": "invalid syntax, expected: <end of file>",
    "labels": [
      {
        "style": "Primary",
        "file_id": null,
        "range": {
          "start": 5,
          "end": 6
        },
        "message": ""
      }
    ],
    "notes": []
  }
]
